starlette/datastructures.py

```diff
@@ -449,11 +449,14 @@ async def read(self, size: int = -1) -> typing.Union[bytes, str]:
             return self.file.read(size)
         return await run_in_threadpool(self.file.read, size)
 
-    async def seek(self, offset: int) -> None:
+    async def seek(self, offset: int, whence: int = 0) -> None:
         if self._in_memory:
-            self.file.seek(offset)
+            self.file.seek(offset, whence)
         else:
-            await run_in_threadpool(self.file.seek, offset)
+            await run_in_threadpool(self.file.seek, offset, whence)
+
+    def tell(self) -> int:
+        return self.file.tell()
```

Comment:
should this one run_in_threadpool if not in memory?

Reply:
I believe _technically_ this should just be reading the position of the file descriptor from memory, but there's nothing that _guarantees_ that this doesn't block, and `aiofiles` itself marks the method as needing to be run in an executor: https://github.com/Tinche/aiofiles/blob/v0.6.0/aiofiles/threadpool/binary.py#L48